### PR TITLE
Issue-41 : Add packages to generate eventlistener and route

### PR DESCRIPTION
**Pull request by chetan-rns**
**What type of PR is this?**

/kind feature

**What does this PR do / why we need it**:

- Added functions and test cases to generate eventlistener
- Added functions and test cases to generate route
- Append the generated resources to ouput slice in bootstrap.go

**Which issue(s) this PR fixes**:
Fixes issue-41

### CODE REVIEW

File: pkg/pipelines/eventlisteners/eventlisteners.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// Filters for interceptors
+const (
+	devCIBuildFilters = "(header.match('X-GitHub-Event', 'pull_request') && body.action == 'opened' || body.action == 'synchronize') && body.pull_request.head.repo.full_name == 'GITHUB_REPO'"
+
+	decCDDeployFilters = "(header.match('X-GitHub-Event', 'push') && body.repository.full_name == 'GITHUB_REPO') && body.ref.startsWith('refs/heads/master')"
+
+	stageCIDryRunFilters = "(header.match('X-GitHub-Event', 'pull_request') && body.action == 'opened' || body.action == 'synchronize') && body.pull_request.head.repo.full_name == 'GITHUB_STAGE_REPO'"
+
+	stageCDDeployFilters = "(header.match('X-GitHub-Event', 'push') && body.repository.full_name == 'GITHUB_STAGE_REPO') && body.ref.startsWith('refs/heads/master')"
+)
```
Comment (bigkevmcd) [inline pkg/pipelines/eventlisteners/eventlisteners.go:19]: The strings that look like this `GITHUB_REPO` and `GITHUB_STAGE_REPO` need to be changed to reflect names generated based on the baseRepo parameter to the `pipelines.Bootstrap` function.

https://github.com/rhd-gitops-example/deployment-pipelines/blob/master/bootstrap.sh#L30-L31

Comment (chetan-rns) [reply pkg/pipelines/eventlisteners/eventlisteners.go:19]: @bigkevmcd PTAL. Now `GITHUB_REPO` and `GITHUB_STAGE_REPO` will be replaced by `baseRepo` and `baseRepo-stage-config` parameters respectively.

File: pkg/pipelines/eventlisteners/eventlisteners.go [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
 
-	stageCDDeployFilters = "(header.match('X-GitHub-Event', 'push') && body.repository.full_name == 'GITHUB_STAGE_REPO') && body.ref.startsWith('refs/heads/master')"
+	stageCDDeployFilters = "(header.match('X-GitHub-Event', 'push') && body.repository.full_name == '%s') && body.ref.startsWith('refs/heads/master')"
 )
 
 // GenerateEventListener will create the required eventlisteners
-func GenerateEventListener() triggersv1.EventListener {
+func GenerateEventListener(baseRepo string) triggersv1.EventListener {
+	githubRepo := baseRepo
+	githubStageRepo := baseRepo + "-stage-config"
```
Comment (bigkevmcd) [inline pkg/pipelines/eventlisteners/eventlisteners.go]: Looks good, you could rename the parameter to `githubRepo` which would drop the need to have the copy?

Comment (chetan-rns) [reply pkg/pipelines/eventlisteners/eventlisteners.go]: Ah..Yes. Renaming the parameter would drop the copy. I'll just rename it.

**Pull request merged**